### ISSUE TITLE
bareos-dbcheck crash whell calling without arguments

### ISSUE THREAD

**Issue by tuxmaster5000** [bareos/bareos#1838]
### Bareos component version

Version 23.0.2

### Steps to reproduce

simple call bareos-dbcheck without any agruments
The call "sudo -u bareos bareos-dbcheck <PGDQL Database name >" will work

### Expected results

An error message or whatever, but now crash

### Actual results

crash with coredump

### Environment

```markdown
- OS: Fedora-39
- component: director database tool bareos-dbcheck
```

### Relevant traces output

```shell
Result of debug via coredumpctl debug:
Core was generated by `bareos-dbcheck'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  __strlen_sse2 () at ../sysdeps/x86_64/multiarch/strlen-sse2.S:142
142             movdqu  (%rax), %xmm4 
(gdb) bt
#0  __strlen_sse2 () at ../sysdeps/x86_64/multiarch/strlen-sse2.S:142
#1  0x00005641198162f9 in std::char_traits<char>::length (__s=0x0) at /usr/include/c++/13/bits/char_traits.h:399
#2  std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::assign (this=0x7ffe57721220, __s=0x0) at /usr/include/c++/13/bits/basic_string.h:1684
#3  0x00005641197eebf8 in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::operator= (__s=<optimized out>, this=0x7ffe57721220)
    at /usr/include/c++/13/bits/basic_string.h:823
#4  main (argc=1, argv=0x7ffe577218d8) at ./src/dird/dbcheck.cc:917
```

**Comment by tuxmaster5000** [comment]
sudo -u bareos bareos-dbcheck -B
will reports correct the data base settings:
```
catalog=Datenbank1
db_name=<database>
db_driver=postgresql
db_user=<user>
db_password=
db_address=
db_port=0
db_socket=
db_type=PostgreSQL
working_dir=/var/lib/bareos

```

**Comment by sebsura** [comment]
@tuxmaster5000 thanks for the report.  I cannot reproduce this on master.  Could you check if this is still an issue for you with a newer release ?
This may have been fixed in #1710 (commit: https://github.com/bareos/bareos/commit/6fb1de8730779ada33a5ffe9b6e1ef5ad0278cc3) already.

**Comment by sebsura** [comment]
Thanks for checking.  I will close this issue for now.  If you run into it on Fedora 39, you can just reopen this issue.
